pyteal/ast/frame.py

```diff
@@ -74,8 +87,16 @@ def has_return(self) -> bool:
 
 class FrameBury(Expr):
     def __init__(self, value: Expr, frame_index: int):
+        from pyteal.ast.subroutine import SubroutineCall
+
         super().__init__()
-        require_type(value, TealType.anytype)
+        if not isinstance(value, SubroutineCall) or not value.output_kwarg:
```

Comment:
Nit: this looks like the 2nd time that we're using this logic. So we can get a little more D.R.Y. with a common helper function. However, feel free to ignore the suggestion if, for example, it's too awkward to consolidate the logic across python packages.

Reply:
the implementation changed by introduction of #593 , resolving